### PR TITLE
Closes #146 - Export/import tasks info as XML/CSV.

### LINKED ISSUE #146: Export/import tasks info as XML/CSV.
While having the ability to create tasks on its own is good,  it is also convenient to export all the working information regarding user's tasks into a file. With this feature implemented, user will have the ability to save the information regarding his work with tasks in .xml and/or .csv files. Also user will able to import task files into his working location.

### PR DESCRIPTION
**Pull request by YarkoKhamar**
I've implemented methods in Tasks service and controller to convert user's tasks to .xml and .csv formats. I've also added two methods to import tasks as a list of Tasks DTO by deserializing  imported from a dialog .xml or .csv file. To serialize and return stream as a response, I've created two custom ActionResults classes which both inherit from FileStream class overriding it's ExecuteResult() method.

### CODE REVIEW

**Comment by roman-fay** [comment]
hold on with merging this.this may require additional discussion

File: ServiceTier/Services/TaskService.cs [hunk 1]
```diff
@@ -150,5 +171,42 @@ public IEnumerable<TaskDTO> GetAllArchivedTasksByUserId(int userId)
...
         }
+
+        public IEnumerable<TaskDTO> ImportTasksFromCsv(IFormFile uploadFile, int userId)
+        {
+            IEnumerable<TaskDTO> listOfTasksDTO = new List<TaskDTO>();
+            if (uploadFile.Length > 0)
+            {
+                using (var stream = uploadFile.OpenReadStream())
+                {
+                    listOfTasksDTO = CsvSerializer.DeserializeFromStream<IEnumerable<TaskDTO>>(stream);
```
Comment (roman-fay) [inline ServiceTier/Services/TaskService.cs]: this should probably be wrapped in try catch

File: ServiceTier/Services/TaskService.cs [hunk 2]
```diff
@@ -150,5 +171,42 @@ public IEnumerable<TaskDTO> GetAllArchivedTasksByUserId(int userId)
...
+
+        public IEnumerable<TaskDTO> ImportTasksFromXml(IFormFile uploadFile, int userId)
+        {
+            IEnumerable<TaskDTO> listOfTasksDTO = new List<TaskDTO>();
+            DefaultXmlSerializer xmlSerializer = new DefaultXmlSerializer(listOfTasksDTO.GetType());
+            if (uploadFile.Length > 0)
+            {
+                using (var stream = uploadFile.OpenReadStream())
+                {
+                    listOfTasksDTO = (IEnumerable<TaskDTO>)xmlSerializer.Deserialize(stream);
```
Comment (roman-fay) [inline ServiceTier/Services/TaskService.cs]: try.. catch

File: ServiceTier/Services/TaskService.cs [hunk 3]
```diff
@@ -150,5 +171,42 @@ public IEnumerable<TaskDTO> GetAllArchivedTasksByUserId(int userId)
 
             return listOfTasksDTO;
         }
+
+        public IEnumerable<TaskDTO> ImportTasksFromCsv(IFormFile uploadFile, int userId)
```
Comment (roman-fay) [inline ServiceTier/Services/TaskService.cs]: what about unit tests for this?

Comment (YarkoKhamar) [reply ServiceTier/Services/TaskService.cs]: Yes, I have a separate task for the unit tests in our project board, which I will complete during this sprint.

File: ServiceTier/Services/TaskService.cs (hunk 1 above)
Comment (roman-fay) [reply ServiceTier/Services/TaskService.cs]: were they added? remind me to talk about those real quick on our meetup

**Pull request merged**